test/e2e-techpreview/onclusterlayering_test.go

```diff
@@ -858,16 +858,28 @@ func assertBuildPodIsAsExpected(t *testing.T, cs *framework.ClientSet, mosb *mcf
 	mcoImages, err := ctrlcommon.GetImagesConfig(context.TODO(), cs.GetKubeclient())
 	require.NoError(t, err)
 
-	buildPod, err := cs.CoreV1Interface.Pods(ctrlcommon.MCONamespace).Get(context.TODO(), mosb.Status.BuilderReference.PodImageBuilder.Name, metav1.GetOptions{})
+	buildJob, err := cs.BatchV1Interface.Jobs(ctrlcommon.MCONamespace).Get(context.TODO(), mosb.Status.BuilderReference.PodImageBuilder.Name, metav1.GetOptions{})
 	require.NoError(t, err)
 
+	buildPods, err := cs.CoreV1Interface.Pods(ctrlcommon.MCONamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: fmt.Sprintf("job-name=%s", buildJob.Name)})
+	require.NoError(t, err)
+
+	var buildPodToCheck corev1.Pod
+	if buildPods != nil {
```

Comment:
issue: What happens if there is more than one build pod? Maybe a guard clause should be added to ensure that there is only one build pod associated with this job name?

Reply:
Fixed to handle multiple pods